src/meta-tx/MetaTransactionUpgradeable.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.0;
+
+import {MinimalForwarderUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/metatx/MinimalForwarderUpgradeable.sol";
+import {ContextMixin} from "./ContextMixin.sol";
+
+contract MetaTransactionUpgradeable is ContextMixin, MinimalForwarderUpgradeable {
+    error FunctionCallFailed();
```

Comment:
```suggestion
    error TransactionFailed();
```

Reply:
This is wrong IMO function call fail is more accurate since the transaction has not failed until we explicitly revert.

Reply:
Agree with Merlin, because only the context can revert without making the whole tx reverting.